src/box/lua/console.lua

```diff
@@ -73,22 +73,20 @@ end
 --
 -- Format a Lua value.
 local function format_lua_value(status, internal_opts, value)
-    local err
-    if status then
-        status, err = pcall(internal.format_lua, internal_opts, value)
-        if status then
-            return err
-        else
-            local m = 'console: exception while formatting output: "%s"'
-            err = m:format(tostring(err))
+    if not status then
+        if value == nil then
+            value = box.NULL
         end
+        value = { error = value }
+    end
+    local res, err = internal.format_lua(internal_opts, value)
+    if res ~= nil then
+        return res
     else
-        err = value
-        if err == nil then
-            err = box.NULL
-        end
+        local m = 'console: exception while formatting the output: "%s"'
+        err = m:format(tostring(err))
+        return internal.format_lua(internal_opts, { error = err })
```

Comment:
> NO_DOC=refactoring
> NO_CHANGELOG=refactoring

It looks like a bug fix, not just refactoring.

> NO_TEST=tested by console_lua.test.lua

`console_lua.test.lua` and all other console tests pass with and without this patch. In fact, if I revert the first three patches, the tests still pass. Can you come up with a test that would cover this code?

Reply:
It's not possible to trigger format_lua failure for !status